### DISCUSSION TITLE
Remove "use client" requirement.

### DISCUSSION THREAD

**Discussion by iolathief108** [vercel/next.js#59482] [Help]
### Summary

is next JS good for large sass application?
Most of the time calculation will be in client side. every single time writing "use client" is annoying.

**Comment by icyJoseph** [comment]
The `use client` directive is an indication for the bundler, that the module has to be bundled for the client/browser.

The usage directive by a given module, implies that, modules imported into this module, are also bundled. So technically, you really only need it at page level.

It doesn't mean though, that the code will only run client side, that's a widespread misconception, which we can thank tech-influencers for.

- https://github.com/reactwg/server-components/discussions/4
